changelog/6316.improvement.rst

```diff
@@ -0,0 +1 @@
+Matching of ``-k EXPRESSION`` to test names is now case-insensitive.
```

Comment:
we must identify this as breaking change

Reply:
We currently don't have a changelog category for that... what do you suggest @RonnyPfannschmidt?

Reply:
creating one

Reply:
I agree. Is there something I can do here?

Reply:
Ronny I suggest we replace "remove" by "breaking", after all every removal is a breaking change, while breaking changes are not necesserily removals

Reply:
:+1:

Reply:
@RonnyPfannschmidt will you do it? I suggest pushing it to this PR already, it is small/trivial set of changes anyway. 

If you don't have the time let me know, I can spare a few minutes to do this if necessary.

Reply:
@nicoddemus i cant get to it today, please sort it out and thanks in advance

Reply:
Done!

Reply:
@RonnyPfannschmidt @BeyondEvil can you guys do a final review?